### PR TITLE
Add a torch.cuda.empty_cache() in utils.save_checkpoint

### PR DESCRIPTION
**Pull request by bfontain**
# What does this PR do?

Adds a empty_cache() to clear the cache right before checkpointing. Some amount of memory is used by NCCL during checkpointing and if we are operating near the total memory of the GPU, empty_cache() can help clear unused memory and allow checkpointing to succeed.

# What issue(s) does this change relate to?

# Before submitting
- [ ] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [ ] Is this change a documentation change or typo fix? If so, skip the rest of this checklist.
- [ ] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [ ] Did you update any related docs and document your change?
- [ ] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [ ] Did you run the tests locally to make sure they pass?
- [ ] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

### CODE REVIEW

File: composer/callbacks/checkpoint_saver.py [hunk 1]
```diff
@@ -14,6 +14,8 @@
 from pathlib import Path
 from typing import Any, Callable, Dict, List, Optional, Union
 
+import torch.cuda
```
Comment (mvpatel2000) [inline composer/callbacks/checkpoint_saver.py:17]: can we just do `import torch` to be simple like elsewhere?

**Review by mvpatel2000 — COMMENTED** [review]
Can we do this in `save_checkpoint` in `checkpoint.py` instead? That way if a user directly calls API, it will also run

File: composer/callbacks/checkpoint_saver.py (hunk 1 above)
Comment (bfontain) [reply composer/callbacks/checkpoint_saver.py]: No longer needed now that the call has been moved to checkpoint.py

**Comment by bfontain** [comment]
> Can we do this in `save_checkpoint` in `checkpoint.py` instead? That way if a user directly calls API, it will also run

Done!

**Review by mvpatel2000 — APPROVED** [review]
Thanks! Might take a bit to merge, CI/CD is broken. I will force merge when tests pass and nothing unusual is failing, please tag me.

**Review by dakinggg — APPROVED** [review]

**Pull request merged**